nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -460,10 +514,16 @@ in {
             ''}
             $CONFIG = [
               'apps_paths' => [
+                ${optionalString (cfg.extraApps != { }) "[ 'path' => '${cfg.home}/nix-apps', 'url' => '/nix-apps', 'writable' => false ],"}
                 [ 'path' => '${cfg.home}/apps', 'url' => '/apps', 'writable' => false ],
                 [ 'path' => '${cfg.home}/store-apps', 'url' => '/store-apps', 'writable' => true ],
               ],
-              'datadirectory' => '${cfg.home}/data',
+              ${if (cfg.appstoreEnable != null)
+                then '''appstoreenabled' => ${lib.boolToString cfg.appstoreEnable},''
+                else (if (cfg.extraApps != { })
+                      then '''appstoreenabled' => false,''
+                      else "")}
```

Comment:
Why don't we set this value at all here? Wouldn't it make sense to use upstream's default (i.e. `true`/`false`) here?

Reply:
Because this may break existing configurations. If users have set `appstoreenabled' => false,` in their own `config.php`, our `override.config.php` will enable their appstore. I opted to only override the property if it is explicitly set by the user (by setting `appstoreEnable` or using `extraApps`)